prov/efa/configure.m4

```diff
@@ -52,23 +52,28 @@ AC_DEFUN([FI_EFA_CONFIGURE],[
 				[efa_happy=0])
 	      ])
 
-	AS_IF([test x"$enable_efa" != x"no"],
-	      [FI_CHECK_PACKAGE([efadv],
-				[infiniband/efadv.h],
-				[efa],
-				[efadv_query_device],
-				[-libverbs],
-				[$efa_PREFIX],
-				[$efa_LIBDIR],
-				[efa_happy=1],
-				[efa_happy=0])
-	      ])
+	AC_ARG_WITH([cuda],
+		    [AC_HELP_STRING([--with-cuda=DIR],
+				    [Provide path to where the CUDA development
+				    and runtime libraries are installed.])],
+		    [FI_CHECK_PACKAGE([cuda],
+				      [cuda_runtime.h],
+				      [cudart],
+				      [cudaMemcpy],
+				      [-lcuda],
+				      [$with_cuda],
+				      [],
+				      [AC_DEFINE([HAVE_CUDA], [1],
+						 [CUDA support])],
+				      [efa_happy=0],
+				      [])],
+		    [])
```

Comment:
I'm not sure if this is the best place to check for cuda availability.  @aingerson - any opinions on this?

Reply:
If I moved it to ofi_cuda.h, I was planning to add it to the main configure.ac.

Reply:
Yeah I would put this check in the main configure and define a global HAVE_LIBCUDA for all providers to use.
I would also lean towards removing the --with-cuda option and just checking for the presence of the libraries/functions and enable HAVE_LIBCUDA if they are present.

Reply:
Hmm, not sure we'd want that. A user might want to have a build that doesn't link to CUDA runtime libraries on a system that does have all the dependencies. Having an explicit flag is better, IMO.  

This also makes it convenient to provide custom paths (`--with-cuda=/usr/local/cuda/`)

Reply:
Users can already provide custom paths to libraries.  I don't like forcing users to know what path a library was installed into.  If we want an option to disable the cuda checks, that would be fine, but I would not go the route of making it difficult in the common use case.

Reply:
Well the user would still have to add the paths to the CFLAGS (I've only ever seen it in /usr/local/cuda) so they could choose to omit it/provide a custom path.
I'll defer to Sean on which option would be best.

Reply:
Sent it before reading Sean's response. I agree with him. If you want to add a --disable-cuda option then I think that's the better option.

Reply:
Yeah, CUDA installs to /usr/local/cuda/ by default and setting CFLAGS/LDFLAGS by hand is worse than giving a configure flag. I'd still prefer doing a AC_ARG_WITH (which can do a without). I'll leave it enabled by default to get the behavior you want.